src/illumidesk/lti13/handlers.py

```diff
@@ -120,3 +123,58 @@ def get(self) -> None:
         # we do not need to use json.dumps because tornado is converting our dict automatically and adding the content-type as json
         # https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.write
         self.write(keys_obj)
+
+
+class FileSelectHandler(BaseHandler):
+    @web.authenticated
+    async def get(self):
+        """Return a sorted list of notebooks recursively found in shared path"""
+        user = self.current_user
+        auth_state = await user.get_auth_state()
+        self.log.debug('Current user for file select handler is %s' % user.name)
+        # decoded = self.authenticator.decoded
+        self.course_id = auth_state['course_id']
+        self.grader_name = f'grader-{self.course_id}'
+        self.grader_root = Path('/home', self.grader_name,)
+        self.course_root = self.grader_root / self.course_id
+        self.course_shared_folder = Path('/shared', self.course_id)
+        a = ''
+        link_item_files = []
+        notebooks = list(self.course_shared_folder.glob('**/*.ipynb'))
+        notebooks.sort()
+        for f in notebooks:
+            fpath = str(f.relative_to(self.course_shared_folder))
+            if fpath.startswith('.') or f.name.startswith('.'):
+                self.log.debug('Ignoring file %s' % fpath)
+                continue
+
+            self.log.debug('Getting files fpath %s' % fpath)
+            url = f'https://{self.request.host}/user/{user.name}/notebooks/shared/{self.course_id}/{fpath}'
+            self.log.debug('URL to fetch files is %s' % url)
```

Comment:
Ok so this is what we would need to change based on what we talked about earlier. Something like:

```
f'https://{self.request.host}/?next=%2Fuser-redirect%2Fgit-pull?repo%3D%252Fhome%252F{user.name}%252Fshared%252F{self.course_id}%26branch%3Dmaster%26urlpath%3Dtree%252F{self.course_id}%252F{fpath}'
```

- Defaults to `shared` folder, since that is the only folder initialized with `git`
- Would this only work if the application is set up in Public mode?

Reply:
Done, I change the url format to use a combination of user-redirect and git-puller. The source repo is the shared directory and the target is placed in the user home directory with the course name. The restriction here is the use of shared-folder feature.